Make Region immutable

rmilfont opened the pull request:
Changes Region class to be immutable.

Commit:Mod

TaskNumber:#60

rmilfont committed:
Make Region immutable

Changes Region class to be immutable.

Commit:Mod

TaskNumber:#60

jmferreiratech commented on src/rbr/Region.java:
```diff
@@ -3,20 +3,20 @@
         this.inputPorts = new HashSet<>(ip);
         this.outputPorts = new HashSet<>(op);
-        this.updateBox();
+        this.box = createBox();
```
This method return something. Do not ask it to do something, instead ask for something. You could pass the destinations to make more sense.

rmilfont replied:
Done.

jmferreiratech commented on src/rbr/Region.java:
```diff
@@ -3,20 +3,20 @@
     }
 
-    private void updateBox() {
+    private Range createBox() {
```
It seems that we have some duplicated code:
https://github.com/LaNoC-UFC/rsbr/blob/develop/src/rbr/RBR.java#L182

rmilfont replied:
Put this method at EdgeColor class. After this, it was renamed to TopologyKnowledge, the idea is that class focus on the knowledge of the topology.

rmilfont committed:
Change EdgeColor class name

Put box method inside EdgeColor and rename to TopologyKnowledge. The idea
is that class focus on knowledge about the topology.

Commit:Mod

TaskNumber:#60
